modules/pre-commit.nix

```diff
@@ -315,7 +329,31 @@ in
                 echo 1>&2 "    3. add .pre-commit-config.yaml to .gitignore"
               else
                 ln -s ${configFile} .pre-commit-config.yaml
-                pre-commit install
+                # Remove any previously installed hooks (since pre-commit iteslf has no idempotent design)
+                hooks="pre-commit pre-merge-commit pre-push prepare-commit-msg commit-msg post-checkout post-commit"
+                for hook in $hooks; do
+                  pre-commit uninstall -t $hook
+                done
+                # Add hooks for configured stages (only) ...
+                if [ ! -z "${concatStringsSep " " cfg.default_stages}" ]; then
```

Comment:
Can you confirm `[]` would evaluate to `""`?

Reason: I did not know how to verify. I failed at getting `lib` into the `nix repl`. :confused:

Reply:
Confirmed.